From 6f8136eb44240ef1d77d02d5119d689183831335 Mon Sep 17 00:00:00 2001
From: Dmitry Smirnov <divis1969@gmail.com>
Date: Mon, 14 Nov 2016 23:30:38 +0300
Subject: Remove CAP_SYS_NICE from surfaceflinger


#---------------------------------------------------------------------------------------------------


diff --git a/libcutils/fs_config.cpp b/libcutils/fs_config.cpp
index 68cfcbe..f192386 100644
--- a/libcutils/fs_config.cpp
+++ b/libcutils/fs_config.cpp
@@ -180,12 +143,8 @@
     { 00750, AID_ROOT,      AID_SHELL,     CAP_MASK_LONG(CAP_SETUID) |
                                            CAP_MASK_LONG(CAP_SETGID),
                                               "system/bin/run-as" },
 
-    // Support FIFO scheduling mode in SurfaceFlinger.
-    { 00755, AID_SYSTEM,    AID_GRAPHICS,  CAP_MASK_LONG(CAP_SYS_NICE),
-                                              "system/bin/surfaceflinger" },
-
     // Support hostapd administering a network interface.
     { 00755, AID_WIFI,      AID_WIFI,      CAP_MASK_LONG(CAP_NET_ADMIN) |
                                            CAP_MASK_LONG(CAP_NET_RAW),
                                               "vendor/bin/hostapd" },

